fl4health/feature_alignment/tab_features_info_encoder.py

```diff
@@ -65,138 +17,114 @@ class TabularFeaturesInfoEncoder:
-        return [self.get_categories()[feature_name] for feature_name in self.features_by_type(ORDINAL)]
+    def get_tabular_features(self) -> List[TabularFeature]:
+        return self.tabular_features
 
-    def get_target(self) -> str:
-        return self.target_info.get_target()
+    def get_tabular_targets(self) -> List[TabularFeature]:
+        return self.tabular_targets
 
-    def get_target_type(self) -> str:
-        return self.target_info.get_target_type()
+    def get_feature_columns(self) -> List[str]:
+        return sorted([feature.get_feature_name() for feature in self.tabular_features])
 
-    def get_target_categories(self) -> List[Scalar]:
-        return self.target_info.get_target_categories()
+    def get_target_columns(self) -> List[str]:
+        return sorted([target.get_feature_name() for target in self.tabular_targets])
 
-    def get_vocabulary(self) -> Dict[str, int]:
-        return self.vocabulary
-
-    def get_all_default_fill_values(self) -> Dict[str, Scalar]:
-        return self.default_fill_values
+    def features_by_type(self, tabular_type: TabularType) -> List[TabularFeature]:
+        return sorted(
+            [feature for feature in self.tabular_features if feature.get_feature_type() == tabular_type],
+            key=TabularFeature.get_feature_name,
+        )
 
-    def get_default_fill_value(self, key: str) -> Scalar:
-        return self.default_fill_values[key]
+    def type_to_features(self) -> Dict[TabularType, List[TabularFeature]]:
+        return {tabular_type: self.features_by_type(tabular_type) for tabular_type in TabularType}
 
-    def set_default_fill_value(self, key: str, value: Scalar) -> None:
-        self.default_fill_values[key] = value
+    def get_categories_list(self) -> List[MetaData]:
+        return [cat_feature.get_metadata() for cat_feature in self.features_by_type(TabularType.ORDINAL)]
 
     def get_target_dimension(self) -> int:
         # Return the dimension of the target array *after* feature alignment is performed.
-        target_type = self.get_target_type()
-        if target_type == ORDINAL or target_type == BINARY:
-            return len(self.get_target_categories())
+        dimension = 0
+        for target in self.tabular_targets:
+            dimension += target.get_metadata_dimension()
+        return dimension
+
+    @staticmethod
+    def _construct_tab_feature(
+        df: pd.DataFrame,
+        feature_name: str,
+        features_to_types: Dict[str, TabularType],
+        fill_values: Optional[Dict[str, Scalar]],
+    ) -> TabularFeature:
+        feature_type = TabularType(features_to_types[feature_name])
+
+        if fill_values is None or feature_name not in fill_values:
+            fill_value = TabularType.get_default_fill_value(feature_type)
+        else:
+            fill_value = fill_values[feature_name]
+
+        if feature_type == TabularType.ORDINAL or feature_type == TabularType.BINARY:
```

Comment:
Are we extracting categories for binary type as well?

Reply:
Yes. This is inherited from the previous iteration of the code because I thought it might be useful if the user wishes to impute missing value with a special unknown value, in which case the binary type would essentially become a categorical type. Do you think it would make sense to keep this?

Reply:
Right, that is something useful to have!